src/Psalm/Internal/Clause.php

```diff
@@ -106,11 +108,15 @@ public function __construct(
         } else {
             ksort($possibilities);
 
-            foreach ($possibilities as $i => $_) {
+            foreach ($possibilities as $i => $v) {
+                if (count($v) < 2) {
```

Comment:
Is this really useful? I'd expect sort to not execute anything if there's nothing to sort

Reply:
I'd expect that to happen too, but alas, this is not what PHP does. Calling sort conditionally is much faster, according to the profiler.

This is something that could be improved in PHP core btw

Reply:
Ok :) Thanks then :)

Reply:
https://github.com/php/php-src/issues/8887